java/ql/src/Security/CWE/CWE-079/AndroidWebViewSettingsEnabledJavaScript.qhelp

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE qhelp PUBLIC
+ "-//Semmle//qhelp//EN"
+ "qhelp.dtd">
+<qhelp>
+  <overview>
+    <p>
+      Enabling JavaScript in an Android WebView allows for the running of JavaScript
+      code in the context of the running application. This opens the possibility for
+      cross-site scripting if the attacker can inject arbitrary JavaScript.
+    </p>
+
+    <p>
+      For example, if your application's WebView allows for visitng web pages
+      which you do not trust, it is possible for an attacker to lead the user to
+      a page which loads malicious JavaScript.
+    </p>
+
+    <p>
+      You can enable or disbale Javascript execution using
+      the <code>setJavaScriptEnabled</code> method of the settings of a WebView.
+    </p>
+  </overview>
+
+  <recommendation>
+    <p>JavaScript execution is disabled by default. You can explicitly disable
+    it by calling <code>setJavaScriptEnabled(false)</code> on the settings of
```

Comment:
Should it be "on the settings" or "of the settings"? The earlier example was "of"

Reply:
I think "on" is appropriate because the sentences work slightly differently.

In `using the <code>setJavaScriptEnabled</code> method of the settings`, I am saying which method is used.

In `by calling <code>setJavaScriptEnabled(false)</code> on the settings`, I am describing a method being used with the settings as the receiver of the action.

Reply:
Got it, thanks for the clarification!